crates/analyzer/src/namespace/items.rs

```diff
@@ -747,6 +772,7 @@ impl ModuleConstantId {
 pub enum TypeDef {
     Alias(TypeAliasId),
     Struct(StructId),
+    Trait(TraitId),
```

Comment:
Ah, yes. That's a good point. I initially made it a type because I started by ignoring generics to just support something like:

```rust
pub fn bar(val: SomeTrait) {
}
```

But then later I noticed that it would be better to add generics first and only support 

```rust
pub fn bar<T: SomeTrait(val: T) {
}
```

I don't know if we ever want to support the first example (or maybe `val: impl SomeTrait` as rust does) or just leave it to generics entirely. My current thinking is that we should not support the first example and leave it to generics to work with traits and hence could remove `Trait` as a type and add it as its own category of `Item`.